azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualNetwork.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import static com.google.common.collect.ImmutableList.copyOf;
+
+import java.util.Map;
+import java.util.List;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class VirtualNetwork {
+
+    @AutoValue
+    public abstract static class AddressSpace{
+
+        public abstract List<String> addressPrefixes();
+
+        @SerializedNames({"addressPrefixes"})
+        public static AddressSpace create(final List<String> addressPrefixes) {
+            return new AutoValue_VirtualNetwork_AddressSpace(copyOf(addressPrefixes));
+        }
+    }
+
+    @AutoValue
+    public abstract static class VirtualNetworkProperties{
+
+        @Nullable
+        public abstract String provisioningState();
+
+        @Nullable
+        public abstract String resourceGuid();
+
+        public abstract AddressSpace addressSpace();
+
+        @Nullable
+        public abstract List<Subnet> subnets();
+
+        @SerializedNames({"provisioningState", "resourceGuid", "addressSpace", "subnets"})
+        public static VirtualNetworkProperties create(final String provisioningState, final String resourceGuid, final AddressSpace addressSpace, final List<Subnet> subnets) {
+
+            return new AutoValue_VirtualNetwork_VirtualNetworkProperties(provisioningState, resourceGuid, addressSpace, subnets != null ? copyOf(subnets) : null);
+        }
+    }
+
+    @Nullable
+    public abstract String name();
+
+    @Nullable
+    public abstract String id();
+
+    @Nullable
+    public abstract String etag();
+
+    public abstract String location();
+
+    public abstract VirtualNetworkProperties properties();
+
+    @Nullable
```

Comment:
@nacx Correct. But unlike other properties, we actually need to support null `tags` parameters in the actual api `create` method since it can be a null input when we create a new resource. This a consistent theme for all the APIs. I have commented on this in the past, maybe it was missed. It is an [optional property that can be empty](https://msdn.microsoft.com/en-us/library/azure/mt163661.aspx) WDYT?